### PR TITLE
Leqiao/add arm64 support

### LINKED ISSUE #10874: Problems with predictions on MacBook Air with M1 chip in Java project based on Maven
I have the following problem:

I'm using ONNX runtime in version `1.8.1` in the Java project with Maven.

When I invoke:

```java
final OrtEnvironment environment = OrtEnvironment.getEnvironment();
```

then I get an error:

```
java.lang.NoClassDefFoundError: Could not initialize class ai.onnxruntime.OrtEnvironment
```

and I cannot perform prediction.

It works fine on Linux on different distros, but I have problems with executing it on MacBook Air with M1 chip.

### PR DESCRIPTION
**Pull request by leqiao-1**
**Description**: add arm64 support

Resolve  #10874
**Motivation and Context**
- add new build artifacts

### CODE REVIEW

**Review by snnn — APPROVED** [review]
LGTM.

**Comment by Craigacp** [comment]
I think the JVM reports the platform as `aarch64` for both Linux and macOS ARM64, so the `osx-arm64` folder should be `osx-aarch64`. You can see our platform detection logic here - https://github.com/microsoft/onnxruntime/blob/master/java/src/main/java/ai/onnxruntime/OnnxRuntime.java#L88. Looks like this chunk (https://github.com/microsoft/onnxruntime/blame/master/cmake/onnxruntime_java.cmake#L81) has been emitting `arm64` for a while, but the loader doesn't know what that is. I'm not sure how I missed that, though the library loading during the tests is a little different to the one from the jar.

I'm not sure what the cmake changes are doing exactly, if those are required for cross compiling then I guess we should modify the loader logic to emit `arm64` rather than `aarch64` when it's detected the OS as `osx`.

**Comment by snnn** [comment]
OSX_ARCHITECTURES is a cmake variable at target level.  It is an array of strings with length of one or two.  It set target specific architectures for macOS. The valid values are:
1. "arm64"
2. "x86_64"
3. "arm64;x86_64", which means arm64 and x86_64. In this case it will generate a binary for both archs. We call such binaries "universal2" binaries. 

If you have set CMAKE_OSX_ARCHITECTURES, it will become the default value of OSX_ARCHITECTURES for every cmake target in you project. 

(The above are not onnxruntime specific. It applies to all cmake projects)

So the lines 81-98 in onnxruntime_java.cmake try to extract arch information from OSX_ARCHITECTURES, then use it to set the JNI_ARCH variable which will be used later in line 136. JNI_ARCH variable determines JAVA_PACKAGE_DIR.

**Comment by Craigacp** [comment]
Ok. I think that there needs to be a chunk in `onnxruntime_java.cmake` like the block at line 95 which sets `JNI_ARCH` to `aarch64`. At that point it should emit the binary in the right place. I'm talking to some people in the Java group to figure out how JNI interacts with universal binaries, but I've not found an answer yet. When I do we might be able to add an extra `osx-universal` target to the loader and have it prefer that for macOS.

**Comment by snnn** [comment]
> I think that there needs to be a chunk in onnxruntime_java.cmake like the block at line 95 which sets JNI_ARCH to aarch64. At that point it should emit the binary in the right place

@[leqiao-1](https://github.com/leqiao-1) please help make the change.

File: cmake/onnxruntime_java.cmake [hunk 1]
```diff
@@ -94,6 +94,8 @@ if(APPLE)
    endif()
    if(JNI_ARCH STREQUAL "x86_64")
        set(JNI_ARCH x64)
+   elseif(JNI_ARCH STREQUAL "arm64")
```
Comment (snnn) [inline cmake/onnxruntime_java.cmake:97]: Put it in "if(APPLE)" branch.

Comment (snnn) [inline cmake/onnxruntime_java.cmake:97]: What does the new package look like?
I suppose you will have "linux-aarch64" and "osx-aarch64"

Comment (snnn) [reply cmake/onnxruntime_java.cmake:97]: BTW, in nuget packages, both of them should be "arm64" instead of "aarch64".

**Review by snnn — APPROVED** [review]

**Pull request merged**